### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by Flink**

### CODE REVIEW

**Review by gschlager — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+## [Unreleased]
+
+## [1.0.0] - 2024-06-14
```
Comment (cvx) [inline CHANGELOG.md]:
```suggestion
## [1.0.0] - 2024-06-17
```

File: .streerc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+--print-width=100
+--plugins=plugin/trailing_comma,disable_ternary
```
Comment (cvx) [inline .streerc]: this is now in stree proper:

```suggestion
--plugins=plugin/trailing_comma,plugin/disable_auto_ternary
```

File: messageformat.gemspec [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    end
+  spec.require_paths = ["lib"]
+
+  spec.add_dependency "mini_racer", ">= 0.6.3"
+
+  spec.add_development_dependency "rake", "~> 13.0"
+  spec.add_development_dependency "minitest", "~> 5.0"
+  spec.add_development_dependency "rubocop-discourse"
+  spec.add_development_dependency "syntax_tree"
+  spec.add_development_dependency "syntax_tree-disable_ternary"
```
Comment (cvx) [inline messageformat.gemspec]: no longer needed!

```suggestion
```

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+inherit_gem:
+  rubocop-discourse: default.yml
+inherit_mode:
+  merge:
+    - Exclude
+Discourse/NoChdir:
+  Enabled: false
+RSpec:
+  Enabled: false
```
Comment (cvx) [inline .rubocop.yml:9]: Are those causing issues? 😃

File: lib/messageformat.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        .new(timeout: 10_000)
+        .tap do |context|
+          context.load(File.expand_path("../dist/messageformat.js", __dir__))
+          context.load(File.expand_path("../dist/compilemodule.js", __dir__))
+          context.eval(<<~JS)
+          function compileMessageFormat(locale, messages) {
+            const mf = new MessageFormat(locale);
+            return compileModule(mf, messages);
+          }
+        JS
```
Comment (cvx) [inline lib/messageformat.rb]: just a bit of whitespace
```suggestion
            function compileMessageFormat(locale, messages) {
              const mf = new MessageFormat(locale);
              return compileModule(mf, messages);
            }
          JS
```

File: dist/compilemodule.js
Comment (cvx) [inline dist/compilemodule.js:1]: Could we gitignore `./dist` and then generate those files in the `publish` step of the workflow? (and make sure those are added in `spec.files` in Gemspec)

Having build artifacts in a repo is always a source of pain (e.g. people always forget to run the build on their end)

File: .rubocop.yml (hunk 4 above)
Comment (Flink) [reply .rubocop.yml:9]: Yes because we’re using minitest specs and not RSpec :sweat_smile:

File: dist/compilemodule.js
Comment (Flink) [reply dist/compilemodule.js:1]: I’ll take a look :)

File: .rubocop.yml (hunk 4 above)
Comment (cvx) [reply .rubocop.yml:9]: ah 😛

File: dist/compilemodule.js
Comment (Flink) [reply dist/compilemodule.js:1]: Done! :grin:

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Lint (Rubocop)
+        run: bundle exec rubocop
+
+      - name: Lint (Syntax Tree)
+        run: bundle exec stree check Gemfile $(git ls-files '*.rb') $(git ls-files '*.rake') $(git ls-files '*.thor')
+
+      - name: Build assets
+        run: |
+          cd messageformat-miniracer
+          yarn install && yarn webpack
```
Comment (cvx) [inline .github/workflows/ci.yml]: dunno if actually shorter, or more concise, or anything but I prefer using `working-directory` 😄 

```suggestion
        working-directory: ./messageformat-miniracer
        run: yarn install && yarn webpack
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Build assets
+        run: |
+          cd messageformat-miniracer
+          yarn install && yarn webpack
```
Comment (cvx) [inline .github/workflows/ci.yml]:
```suggestion
        working-directory: ./messageformat-miniracer
        run: yarn install && yarn webpack
```

**Review by cvx — APPROVED** [review]
Looks great! 🚀

**Pull request merged**